Nodejs/Product/Npm/SemverVersion.cs

```diff
@@ -31,13 +31,13 @@ public struct SemverVersion {
             "^(?<major>[0-9]+)"
             + "\\.(?<minor>[0-9]+)"
             + "\\.(?<patch>[…0-9]+)" // The '…' is there to handle the 'classy' library, which has a very long version number - see slightly snarky comment about that unadulterated bag of hilarity below.
-            + "(?:-(?<prerelease>[…0-9A-Za-z-]+(\\.[…0-9A-Za-z-]+)*))?"
-            + "(?:\\+(?<buildmetadata>[…0-9A-Za-z-]+(\\.[…0-9A-Za-z-]+)*))?$",
-            RegexOptions.IgnoreCase | RegexOptions.Singleline);
+            + "(?:-(?<prerelease>[…0-9A-Za-z-.]+))?"

```

Comment:
I get adding the "." to allow multi-part pre-release tags (i.e. "1.0.1-beta.1"), but doesn't this also allow invalid formats such as "1.0.1-..5.."? Also, I though the "-" had to be immediately after the opening "[" or before the closing "]" to be treated as a regular character rather than the range operator.